sky/resources.py

```diff
@@ -221,6 +221,9 @@ def __init__(
 
         self._cluster_config_overrides = _cluster_config_overrides
 
+        self._use_az_ml = skypilot_config.get_nested(('azure', 'use_az_ml'),
+                                                     False)
```

Comment:
Adding a cloud-specific setting in `Resources` is not clean. Can we get rid of it?